cognite/cdffs/spec.py

```diff
@@ -719,19 +746,36 @@ def strategy_submit(data: bytes, index: int) -> None:
 
         # If it's the final block, then send a merge request
         if final:
-            total_size = self.write_strategy.merge_chunks()
 
-            self.cognite_client.files.update(
-                item=FileMetadataUpdate(external_id=self.external_id).metadata.add({"size": total_size})
+            @retry(
+                stop=stop_after_attempt(5),
+                wait=wait_fixed(0.5),
+                after=after_log(logging.getLogger("cdffs"), logging.INFO),
             )
+            def safe_merge() -> int:
```

Comment:
Do we really need this nested function here?

Reply:
In case something fails inside the strategy merge it actually can be retried

Reply:
tenacity comes with attributes only, so having smaller retry scope is better, I think